hypertrace-view-generator/hypertrace-view-generator/src/main/java/org/hypertrace/viewgenerator/generators/SpanEventViewGenerator.java

```diff
@@ -140,21 +143,93 @@ private Event getApiEntrySpanForExitSpan(
    * api_node to api_entry_span (child of api_exit_span) in another api_node 2. exit calls to
    * backend from api_exit_span in api_node
    */
-  Map<ByteBuffer, Integer> computeApiExitCallCount(StructuredTrace trace) {
+  Map<ByteBuffer, ApiExitCallInfo> computeApiExitCallCount(StructuredTrace trace) {
```

Comment:
I remember asking before, but don't remember the answer and have the same question every time I see more code being added to view gen - why isn't this an enricher? There was _never_ supposed to be logic in view generation and it's one thing to defer fixing that, but why add more?

Reply:
Hey, I did refactor ApiTraceGraph to move some repeated business logic from view-generators to ApiTraceGraph as part of https://github.com/hypertrace/hypertrace-ingester/issues/139 (with the intention to achieve what you suggest, but stopped it after a bit of refactoring)
But in the current state I don't see the possibility to move the entire stuff as enrichers in trace-enricher, unless we publish on the view topics itself from enrichers. 
Will do an offline discussion on what blockers I see, but at the same time, we do have to move ahead with the feature requirement while balancing the refactoring work.

Reply:
I am not sure about the complexity required. 

But, can't we enrich what you are doing here (_by copying the same logic in enricher_) as a map in the event? `Map<String, Integer> calleeNameToExitCallCount`

I believe we do support adding map as an enriched attribute

Reply:
Added it as an enricher.